### PR TITLE
impl(sql): enable integration tests

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11511)

### CODE REVIEW

File: google/cloud/sql/integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -53,3 +53,5 @@ function (google_cloud_cpp_sql_define_integration_tests)
         return()
     endif ()
 endfunction ()
+
+google_cloud_cpp_sql_define_integration_tests()
```
Comment (devbww) [inline google/cloud/sql/integration_tests/CMakeLists.txt:57]: Does this need to be guarded by an `if (BUILD_TESTING)` conditional?  (All similar ones seem to be.)

Comment (dbolduc) [reply google/cloud/sql/integration_tests/CMakeLists.txt:57]: > Does this need to be guarded

it is: https://github.com/googleapis/google-cloud-cpp/blob/8b4fc2f96604b704b2629dc7a5e454d34af4e533/google/cloud/sql/CMakeLists.txt#L126

---

We don't need to define a `function`, then call it. We can just run the things inside.

**Review by dbolduc — APPROVED** [review]

File: google/cloud/sql/integration_tests/CMakeLists.txt (hunk 1 above)
Comment (devbww) [reply google/cloud/sql/integration_tests/CMakeLists.txt:57]: OK, thanks.  I'll just note that sometimes we have `add_subdirectory(integration_tests)` being conditional, and sometimes not.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11511)* status: 0 of 2 files reviewed, all discussions resolved (waiting on @dbolduc and @devbww)

___
*[`google/cloud/sql/integration_tests/CMakeLists.txt` line 57 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11511#gh-1186348372:-NUh2qXI4UHl2W7C6Xpc:b8af9ec) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/4f58e017dd0d415ebf623408108cf35d44744d3f/google/cloud/sql/integration_tests/CMakeLists.txt#L57)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

OK, thanks.  I'll just note that sometimes we have `add_subdirectory(integration_tests)` being conditional, and sometimes not.
</blockquote></details>

Removed function definition plus call.

**Pull request merged**